lib/modules.nix

```diff
@@ -236,6 +259,9 @@ let
             moduleType = type;
           };
           _module.specialArgs = specialArgs;
+          # This is not actually assigned
+          # 'apply' discards the value but keeps the definition
```

Comment:
This line doesn't seem to be part of the problem, so I don't think we need to explain it here.
If the workaround is fixed properly this line can probably stick around.
Also I expect user expectations about this option to be met by the workaround anyway.
```suggestion
```